src/publishSubmit.ml

```diff
@@ -118,78 +103,39 @@ module GH = struct
         OpamFilename.remove tok_file;
         get_user_token root (repo_owner, repo_name)
     else
-    let user, token =
-      if
-        OpamConsole.confirm
-          "No Github token found. Should I generate one ?\n\
-           (your credentials will be required. If you do not have a Github \
-           account, you can create one at https://github.com/join)"
-      then
-        let hostname = Unix.gethostname () in
-        let token_note = token_note hostname in
-        OpamConsole.msg
-          "The obtained token will be stored in %s.\n\
-           Your active tokens can be seen and revoked at \
-           https://github.com/settings/tokens\n\n"
-          (OpamFilename.prettify tok_file);
-        let user =
-          let rec get_u () =
-            match OpamConsole.read "Please enter your Github name:" with
-            | Some u -> u
-            | None -> get_u ()
-          in
-          get_u ()
+    let token =
+      OpamConsole.msg
+        "In order to continue, you will be required to provide your GitHub \
+         personal access token. This personal acces token requires the \
+         \"public_repo\" scope (or \"repo\" if you submit to a private \
+         repository).\n\
+         You can manage your personal access token at \
+         https://github.com/settings/tokens\n\
+         Note: your GitHub password will not work, only a personal access token \
+         can authenticates you.\n\n";
```

Comment:
People tend to skip long messages, I think it's better to provide the link as early as possible since that's the action to take.
Shorter and to the point:

```suggestion
        "Please generate a Github token at \
         https://github.com/settings/tokens/new to allow access.\n\
         The \"public_repo\" scope is required (\"repo\" if submitting to a \
         private opam repository).\n"
```